src/rgw/rgw_rest.cc

```diff
@@ -1894,6 +1894,8 @@ int RGWHandler_REST::read_permissions(RGWOp* op_obj)
     /* is it a 'create bucket' request? */
     if (op_obj->get_type() == RGW_OP_CREATE_BUCKET)
       return 0;
+    if (op_obj->get_type() == RGW_OP_GET_OBJ) break;
+    
     only_bucket = true;
```

Comment:
this is an object operation, nit a bucket one. setting the only_bucket flag to "true" could be mislieading

Reply:
it not setting the only_bucket.
otherwise it failed the permission

Reply:
i understand, but at least rename the variable so people would know what it means. also, make sure that other places that use it don;t assume something on the request to have no "object" part in it

Reply:
i'm not sure i follow. 
i did not declare /**only_bucket**/ variable and i dont using it.

Reply:
trying to understand. before your change, you would get into the `OP_COPY` case (BTW, why?), and `only_bucket` would set to "true".
after you change, it remain "false" because of the `break`.
ok, this makes sense, as this is an object and not a bucket operation.

Reply:
it get into OP_POST.
its a post request which actually do get-object.

Reply:
ok, i see now. maybe, for better clarity, override get_type() with a new value: `RGW_OP_SELECT_OBJ` ?
this would make sure that other combinations of POST+GET don't fall into this case?

Reply:
not sure about that (but we can check that option). 
i made 2 minor changes, i added case to switch-case(as done by others) , for v4-authentications and permissions.

i used inheritance and "the base original setting", i did not override it,  with the thought to change only what's necessary. 
actually (with my last changes), the PR only override /send_response_data/ , all the rest of the operations done in base, with no modifications. 
and that's the main operation of s3-select , it modify the response after getting the object.

Reply:
the code here is about authentication. you wan to identify the s3select messages so that the correct authentication would happen.
bets way to identify them, is explicitly according to their type (derived from existence of the "select-type" arg.
here you identify them as POST ops + GET, may cause issues if another messages has this combination

Reply:
lets discuss this offline.
my main points as follows
*) the new derived object , re-use almost all operations , except of the send-response-data (its only new action)
*) by changing/overriding the get-type , it might impact other current flows or future flows as get-obj is evolving.
*) its possible to distinct the s3-select , by adding get-sub-type , or by adding a distinction in condition.

Reply:
i would recommend using `is_obj_update_op()` to cover this case. add a `RGWHandler_REST_Obj_S3::is_select_op()` and append `|| is_select_op()` to the end of `is_obj_update_op()`